### PR TITLE
HPCC-14055 Error from regression suite test "readresult.ecl"

### PR DESCRIPTION
**Pull request by richardkchapman**
This highlighted a couple of regressions introduced into workunitservices by
prior fixes.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: plugins/workunitservices/workunitservices.cpp [hunk 1]
```diff
@@ -433,7 +433,7 @@ WORKUNITSERVICES_API void wsWorkunitList(
         addWUQueryFilter(filters, filterCount, filterbuf, fileread, (WUSortField) (WUSFfileread | WUSFnocase));
         addWUQueryFilter(filters, filterCount, filterbuf, filewritten, (WUSortField) (WUSFfilewritten | WUSFnocase));
         addWUQueryFilter(filters, filterCount, filterbuf, username, (WUSortField) (WUSFuser | WUSFnocase));
-        addWUQueryFilter(filters, filterCount, filterbuf, jobname, (WUSortField) (WUSFjob | WUSFnocase));
+        addWUQueryFilter(filters, filterCount, filterbuf, jobname, (WUSortField) (WUSFjob | WUSFwild | WUSFnocase));
```
Comment (jakesmith) [inline plugins/workunitservices/workunitservices.cpp]: This was the only actual change right in this commit?
The other changes (varApppend -> varAppendMax) were just a name change ?

**Comment by jakesmith** [comment]
Looks fine, assuming I haven't missed something (re. question above)
@richardkchapman

File: plugins/workunitservices/workunitservices.cpp (hunk 1 above)
Comment (richardkchapman) [inline plugins/workunitservices/workunitservices.cpp]: No, there was some confusion about the varAppend vs varAppendMax overloading, and the wrong function was being called. The name change was done to remove that confusion.

File: plugins/workunitservices/workunitservices.cpp [hunk 2]
```diff
@@ -318,10 +318,10 @@ static bool addWUQueryFilter(WUSortField *filters, unsigned &count, MemoryBuffer
 static bool serializeWUInfo(IConstWorkUnitInfo &info,MemoryBuffer &mb)
 {
     fixedAppend(mb,24,info.queryWuid());
-    varAppend(mb,64,info.queryUser());
-    varAppend(mb,64,info.queryClusterName());
-    varAppend(mb,64,""); // roxiecluster is obsolete
-    varAppend(mb,256,info.queryJobName());
+    varAppendMax(mb,64,info.queryUser());
```
Comment (richardkchapman) [inline plugins/workunitservices/workunitservices.cpp]: These changes were more than just a name change (wrong function called)

**Comment by jakesmith** [comment]
@richardkchapman - good to merge

**Pull request merged**